lib/beaker-puppet/install_utils/foss_utils.rb

```diff
@@ -1092,21 +1092,21 @@ def install_packages_from_local_dev_repo( host, package_name )
         # @return nil
         def install_puppet_agent_dev_repo_on( hosts, opts )
 
-          opts[:puppet_agent_version] ||= opts[:version] #backward compatability
-          if not opts[:puppet_agent_version]
-            raise "must provide :puppet_agent_version (puppet-agent version) for install_puppet_agent_dev_repo_on"
-          end
-          # TODO consolidate these values as they serve no purpose from beaker's side
-          # you could provide any values you could to one to the other
-          puppet_agent_version = opts[:puppet_agent_sha] || opts[:puppet_agent_version]
+          block_on hosts do |host|
+            opts[:puppet_agent_version] ||= opts[:version] #backward compatability
+            if not opts[:puppet_agent_version]
+              raise "must provide :puppet_agent_version (puppet-agent version) for install_puppet_agent_dev_repo_on"
+            end
```

Comment:
Since this part operates on the `opts` passed into the method, I think it should be outside of the `hosts` loop.

Might also rename the `opts` variable below so that it doesn't shadow the parameter above.